frontend/src/modules/Folders/views/FolderView.js

```diff
@@ -233,11 +233,11 @@ const FolderView = () => {
   const fetchItem = useCallback(async () => {
     setLoading(true);
     const response = await client.query(getDatasetStorageLocation(params.uri));
-    if (!response.errors && response.data.getDatasetStorageLocation !== null) {
+    if (response.data.getDatasetStorageLocation !== null) {
       setFolder(response.data.getDatasetStorageLocation);
       setIsAdmin(
         ['Creator', 'Admin', 'Owner'].indexOf(
-          response.data.getDatasetStorageLocation.dataset.userRoleForDataset
+          response.data.getDatasetStorageLocation.dataset?.userRoleForDataset
```

Comment:
Should we dispatch all errors when the data is null?  or can we assume that the first error will correspond to the top-level query (I think so right? graphql will execute first the top resolver and then the field-resolvers)
- here in FolderView we are just dispatching the first error
- in DatasetListView we dispatch all errors with the forEach loop

Reply:
Good catch, my intention was to dispatch all everywhere, will change it